test/ha/ha.go

```diff
@@ -26,34 +26,48 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apimachinery/pkg/util/wait"
+
 	"knative.dev/pkg/test"
 	"knative.dev/pkg/test/logging"
 )
 
+func countingRFind(wr rune, wc int) func(rune) bool {
+	cnt := 0
+	return func(r rune) bool {
+		if r == wr {
+			cnt++
+		}
+		return cnt == wc
+	}
+}
+
+func extractDeployment(pod string) string {
+	if x := strings.LastIndexFunc(pod, countingRFind('-', 2)); x != -1 {
+		return pod[:x]
+	}
+	return ""
+}
+
 // GetLeaders collects all of the leader pods from the specified deployment.
-func GetLeaders(t *testing.T, client *test.KubeClient, deploymentName, namespace string) ([]string, error) {
+func GetLeaders(t *testing.T, client *test.KubeClient, deploymentName, namespace string) (sets.String, error) {
```

Comment:
This is a breaking change.  `len(leaders)` on line 88 may never be satisfied if this deduplicates

Reply:
Done